src/compiler/parser.ts

```diff
@@ -1333,58 +1334,80 @@ module ts {
         }
 
         function parseComputedPropertyName(): ComputedPropertyName {
-            // PropertyName[Yield,GeneratorParameter] :
+            // PropertyName[Yield,GeneratorParameter,Await,AsyncParameter] : (Modified)
             //     LiteralPropertyName
             //     [+GeneratorParameter] ComputedPropertyName
-            //     [~GeneratorParameter] ComputedPropertyName[?Yield]
+            //     [+AsyncParameter] ComputedPropertyName

```

Comment:
Currently, yes.

From: Jason Freeman [mailto:notifications@github.com]
Sent: Thursday, February 12, 2015 2:44 PM
To: Microsoft/TypeScript
Cc: Ron Buckton
Subject: Re: [TypeScript] Async Functions (#1781)


In src/compiler/parser.ts<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24628267>:

>              //     LiteralPropertyName

>              //     [+GeneratorParameter] ComputedPropertyName

> -            //     [~GeneratorParameter] ComputedPropertyName[?Yield]

> +            //     [+AsyncParameter] ComputedPropertyName

So yield and await are mutually exclusive contexts, right?

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24628267>.